File: ch3/segflt_nocore.c
#define _GNU_SOURCE
#include <stdlib.h>
#include <string.h>
#include <sys/types.h>
#include <sys/resource.h>
#include "../common.h"

int main(void) {
	char *buf = malloc(3);
	char *sgflt = buf + (10000000 * sizeof(char));
	struct rlimit rlim, rlim_old;
	rlim.rlim_cur = rlim.rlim_max = (rlim_t) 0;
	if (prlimit(0, RLIMIT_CORE, &rlim, &rlim_old) == -1)
		FATAL("prlimit:cpu failed\n");
	memcpy(sgflt, buf, 3);
}

File: ch3/segflt_nocore.zig
const std = @import("std");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    var allocator = gpa.allocator();
    var buf = try allocator.alloc(u8, 3);
    defer allocator.free(buf);
    var buf_ptr = buf.ptr;
    var sgflt_ptr = buf_ptr + (10000000 * @sizeOf(u8));
    var rlim = std.os.rlimit{ .cur = 0, .max = 0 };
    if (std.os.linux.prlimit(0, std.os.rlimit_resource.CORE, &rlim, null) == -1) {
        std.debug.panic("prlimit:cpu failed\n", .{});
    }
    std.mem.copy(u8, sgflt_ptr[0..3], buf);
}
